src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -86,6 +86,7 @@ ZeroInferRequest::ZeroInferRequest(const std::shared_ptr<ZeroInitStructsHolder>&
     _logger.debug("ZeroInferRequest::ZeroInferRequest - checking level zero attributes and allocating tensors");
 
     size_t ioIndex = 0;
+    auto batchSize = _graph->get_batch_size(_metadata, _userInputTensors.at(0));
```

Comment:
I'm mostly sure that we don't need to determine batch_size here  in order to allocate batched tensor on L0 side for dynamic batch scenario as we don;t know a real batch size in truly dynamic case.
We may only allocate an upper boundary, but it seems kinda premature optimizations(?), because we WILL reallocate pipeline before an actual infer happens after user had specified all tensors